libs/tools/generator/core/src/services/generator-metadata-provider.ts

```diff
@@ -0,0 +1,250 @@
+      }
+
+      algorithms = AlgorithmsByType[requested.type].concat(forwarders);
+    } else if (requested.algorithm && isForwarderExtensionId(requested.algorithm)) {
+      algorithms = this.site.extensions.has(requested.algorithm.forwarder)
+        ? [requested.algorithm]
+        : [];
+    } else if (requested.algorithm) {
+      algorithms = Algorithms.includes(requested.algorithm) ? [requested.algorithm] : [];
+    } else {
+      this.log.panic(requested, "algorithm or category required");
+    }
+
+    return algorithms;
+  }
+
+  // emits a function that returns `true` when the input algorithm is available
+  private isAvailable$(
+    dependencies: BoundDependency<"account", Account>,
+  ): Observable<(a: CredentialAlgorithm) => boolean> {
+    const id$ = dependencies.account$.pipe(
+      map((account) => account.id),
+      pin(),
+      shareReplay({ bufferSize: 1, refCount: true }),
+    );
+
+    const available$ = id$.pipe(
+      switchMap((id) => {
+        const policies$ = this.application.policy.getAll$(PolicyType.PasswordGenerator, id).pipe(
+          map((p) => availableAlgorithms_vNext(p).filter((a) => this._metadata.has(a))),
+          map((p) => new Set(p)),
+          // complete policy emissions otherwise `switchMap` holds `available$` open indefinitely
+          takeUntil(anyComplete(id$)),
+        );
+        return policies$;
+      }),
+      map(
+        (available) =>
+          function (a: CredentialAlgorithm) {
+            return isForwarderExtensionId(a) || available.has(a);
+          },
+      ),
+    );
+
+    return available$;
+  }
+
+  /** Retrieve credential algorithms filtered by the user's active policy.
+   *  @param requested when this has a `category` property, the method
+   *   returns all algorithms in the category. When this has an `algorithm`
+   *   property, the method returns 0 or 1 matching algorithms.
+   *  @param dependencies.account the account requesting algorithm access;
+   *    this parameter controls which policy, if any, is applied.
+   *  @returns an observable that emits matching algorithms. When no algorithms
+   *    match the request, an empty array is emitted.
+   *  @throws when neither `requested.algorithm` nor `requested.category` contains
+   *    a value.
+   *  @remarks this method applies policy controls. In particular, it excludes
+   *    algorithms prohibited by a policy control. If you want lists of algorithms
+   *    supported by the client, use `algorithms`.
+   */
+  algorithms$(
+    requested: AlgorithmRequest,
+    dependencies: BoundDependency<"account", Account>,
+  ): Observable<CredentialAlgorithm[]>;
+  algorithms$(
+    requested: TypeRequest,
+    dependencies: BoundDependency<"account", Account>,
+  ): Observable<CredentialAlgorithm[]>;
+  algorithms$(
+    requested: MetadataRequest,
+    dependencies: BoundDependency<"account", Account>,
+  ): Observable<CredentialAlgorithm[]> {
```

Comment:
❓ Same question for these types

```suggestion
  algorithms$(
    requested: AlgorithmRequest | TypeRequest | MetadataRequest,
    dependencies: BoundDependency<"account", Account>,
  ): Observable<CredentialAlgorithm[]> {
```